common/src/main/java/org/embeddedt/modernfix/common/mixin/perf/cache_profile_texture_url/YggdrasilGsonDeserializerMixin.java

```diff
@@ -0,0 +1,34 @@
+package org.embeddedt.modernfix.common.mixin.perf.cache_profile_texture_url;
+
+import com.mojang.authlib.minecraft.MinecraftProfileTexture;
+import com.mojang.authlib.yggdrasil.YggdrasilMinecraftSessionService;
+
+import com.llamalad7.mixinextras.injector.wrapoperation.Operation;
+import com.llamalad7.mixinextras.injector.wrapoperation.WrapOperation;
+import org.embeddedt.modernfix.annotation.ClientOnlyMixin;
+import org.embeddedt.modernfix.duck.ICachedProfileTexture;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+
+@Mixin(value=YggdrasilMinecraftSessionService.class, remap=false)
```

Comment:
I don't think an authlib mixin will be applied on Forge since authlib is a library. How about computing the cached hash on the first call to `getHash` instead? It should be done in a local variable to prevent race conditions.

```java
getHash() {
    String hash = this.savedHash;
    if(hash == null) {
        this.savedHash = hash = <do hash computation>;
    }
    return hash;
}
```

This should also simplify the code since you don't need a second mixin anymore.